ragna/deploy/_ui/components/metadata_filters_builder.py

```diff
@@ -0,0 +1,307 @@
+                        return False
+                return True
+            else:
+                value = values[0]
+
+        if METADATA_FILTERS[self.key]["type"] == int:
+            try:
+                int(value)
+            except ValueError:
+                return False
+        elif METADATA_FILTERS[self.key]["type"] == datetime:
+            try:
+                datetime.strptime(value, "%Y-%m-%d %H:%M:%S")
+            except ValueError:
+                return False
+
+        return True
+
+    def validate(self):
+        def change_valid_status(valid, widget):
+            if valid:
+                widget.css_classes = [
+                    c for c in widget.css_classes if c != "metadata-filter-error"
+                ]
+            else:
+                widget.css_classes = widget.css_classes + ["metadata-filter-error"]
+
+        if self.key == "":
+            change_valid_status(False, self.key_select)
+            return False
+
+        change_valid_status(True, self.key_select)
+
+        result = True
+        if self.operator == "":
+            change_valid_status(False, self.operator_select)
+            result = False
+        else:
+            change_valid_status(True, self.operator_select)
+
+        if not self.validate_value(self.value):
+            change_valid_status(False, self.value_text_input)
+            result = False
+        else:
+            change_valid_status(True, self.value_text_input)
+
+        return result
+
+    def __str__(self):
+        return f"{self.key} {self.operator} {self.value}"
+
+    def __repr__(self):
+        return f"[key:{self.key} \t op:{self.operator} \t val:{self.value}]"
+
+    def convert_operator(self, operator):
+        if operator == "==":
+            return MetadataFilter.eq
+        if operator == "!=":
+            return MetadataFilter.ne
+        if operator == ">":
+            return MetadataFilter.gt
+        if operator == "<":
+            return MetadataFilter.lt
+        if operator == ">=":
+            return MetadataFilter.ge
+        if operator == "<=":
+            return MetadataFilter.le
+        if operator == "in":
+            return MetadataFilter.in_
+        if operator == "not in":
+            return MetadataFilter.not_in
+
+        # if operator == "raw":
+        #    return MetadataFilter.RAW
+
+    def convert_value(self, value):
+        try:
+            if METADATA_FILTERS[self.key]["type"] == int:
+                return int(value)
+            elif METADATA_FILTERS[self.key]["type"] == datetime:
+                return datetime.strptime(value, "%Y-%m-%d %H:%M:%S")
+            return value
+        except ValueError:
+            return value
+
+    def get_metadata_filter(self):
+        if self.key == "" or self.operator == "" or self.value == "":
+            return None
+        return self.convert_operator(self.operator)(
+            key=self.key, value=self.convert_value(self.value)
+        )
```

Comment:
Instead of letting `convert_operator` return the convenience method, let's have it return the actual operator and instantiate the `MetadataFilter` manually:

```suggestion
        return MetadataFilter(
            operator=self.convert_operator(self.operator),
            key=self.key,
            value=self.convert_value(self.value),
        )
```